### PR TITLE
add no-notifications flag for backup:run command, fixes #461

### LINKED ISSUE #461: Suppress email notification via command line
Hi,

I'm using the Laravel schedule() to schedule hourly database backups.  I'm using the 'backup:run --only-db' command.  Is there a way I can specify not to generate a notification for this?  e.g. 'backup:run --only-db --ignore-notification'.

I like receiving notifications when I do normal backups 'backup:run', but the hourly notification is a bit much.  Thanks!

### PR DESCRIPTION
**Pull request by fetzi**
add the `--no-notifications` flag as mentioned in #461

### CODE REVIEW

**Comment by freekmurze** [comment]
All commands could benefit from the added option. But maybe it's better than you hold of on that until I code review this one. Will do that in the next few days.

File: src/Commands/BackupCommand.php [hunk 1]
```diff
@@ -10,7 +10,7 @@
 class BackupCommand extends BaseCommand
 {
     /** @var string */
-    protected $signature = 'backup:run {--filename=} {--only-db} {--only-files} {--only-to-disk=}';
+    protected $signature = 'backup:run {--filename=} {--only-db} {--only-files} {--only-to-disk=} {--no-notifications}';
```
Comment (freekmurze) [inline src/Commands/BackupCommand.php]: Rename the option to `--disable-notifications`

File: src/Commands/BackupCommand.php [hunk 2]
```diff
@@ -19,6 +19,8 @@ public function handle()
     {
         consoleOutput()->comment('Starting backup...');
 
+        $noNotifications = $this->option('no-notifications');
```
Comment (freekmurze) [inline src/Commands/BackupCommand.php]: Rename to `$disableNotifications`

File: src/Tasks/Backup/BackupJob.php [hunk 3]
```diff
@@ -32,6 +32,9 @@ class BackupJob
     /** @var \Spatie\TemporaryDirectory\TemporaryDirectory */
     protected $temporaryDirectory;
 
+    /** @var bool */
+    protected $triggerNotifications = true;
```
Comment (freekmurze) [inline src/Tasks/Backup/BackupJob.php]: Rename to `sendNotifications`

File: src/Tasks/Backup/BackupJob.php [hunk 4]
```diff
@@ -127,7 +137,7 @@ public function run()
         } catch (Exception $exception) {
             consoleOutput()->error("Backup failed because {$exception->getMessage()}.".PHP_EOL.$exception->getTraceAsString());
 
-            event(new BackupHasFailed($exception));
+            $this->triggerNotification(new BackupHasFailed($exception));
```
Comment (freekmurze) [inline src/Tasks/Backup/BackupJob.php]: rename to `sendNotification`

**Comment by fetzi** [comment]
@freekmurze should I implement this feature for `cleanup` and `monitor` also in this PR?

**Comment by freekmurze** [comment]
For cleanup ok, for monitor it seems unnecessary as the primary function for that command is sending notifications.

**Comment by freekmurze** [comment]
Thank you very much!

**Pull request merged**